inspirehep/modules/orcid/cache.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+from flask import current_app as app
+import hashlib
+from redis import StrictRedis
+from StringIO import StringIO
+
+from .converter import OrcidConverter
+
+
+class OrcidCache(object):
+    def __init__(self, orcid):
+        redis_url = app.config.get('CACHE_REDIS_URL')
+        self.redis = StrictRedis.from_url(redis_url)
+        self.orcid = orcid
+
+    def _get_key(self, recid):
+        """Return the string 'orcidcache:``orcid_value``:``recid``'"""
+        return 'orcidcache:{}:{}'.format(self.orcid, recid)
+
+    def write_record_data(self, recid, putcode, hash_value=None):
+        """
+        Write the putcode and the hash for the given (orcid, recid).
+
+        Args:
+            recid(int): inspire record's id.
+            putcode(string): the putcode used to push the record to ORCID.
+            hash_value(Optional[string]): hashed ORCID record content.
+        """
+        key = self._get_key(recid)
+        value = {'putcode': putcode}
+        if hash_value:
+            value['hash'] = hash_value
+        self.redis.hmset(key, value)
+
+    def read_record_data(self, recid):
+        """
+        Read the putcode and the hash for the given (orcid, recid).
+
+        Args:
+            recid(int): inspire record's id.
+        """
+        key = self._get_key(recid)
+        value = self.redis.hgetall(key)
+        return value.get('putcode'), value.get('hash')
+
+
+class OrcidHasher(object):
+    def compute_hash(self, record):
```

Comment:
As mentioned above, why not make it a classmethod, `OrcidHasher` doesn't seem to have any state, no?

Reply:
it needs `self` to find the other method it's calling. I agree on your distate of this style though :)

Reply:
It is a class as a mean of grouping relate methods. In particular it is a simple template design pattern, often implemented in Python the way I did, see this f.e.:
https://sourcemaking.com/design_patterns/template_method/python/1

`compute_hash()` can be a class method yes, but I prefer to keep it as instance method with a classic signature